### ISSUE TITLE
[FIRRTL] Add folders for IntegerShrOp

### ISSUE THREAD

**Issue by mikeurbach** [llvm/circt#6725]
This is a primitive Property operation, defined by FIRRTL spec section 25.1.3. General folds for signed arbitrary precision integer arithmetic can be applied, but none were defined in the initial PR. This could include things like:

* [ ] integer_shr(a, 0) -> a
* [ ] integer_shr(constantA, constantB) -> constantA >> constantB
* [ ] etc...

**Comment by stomfaig** [comment]
Is this issue still active? If so I am happy to give it a go.

**Comment by mikeurbach** [comment]
Yep, I don't think anyone looked into this one, so feel free if you're interested!
